File: C# Fundamentals/BasicSyntax/VendingMachine.cs
using System;

namespace VendingMachine
{
    class Program
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine().ToLower();

            double cash = 0;
            while (input != "start")
            {
                if (Convert.ToDouble(input) == 0.1 || 
                    Convert.ToDouble(input) == 0.2 || 
                    Convert.ToDouble(input) == 0.5 || 
                    Convert.ToDouble(input) == 1.0 || 
                    Convert.ToDouble(input) == 2.0)
                {
                    cash += Convert.ToDouble(input);
                }
                else
                {
                    Console.WriteLine($"Cannot accept {input}");
                }

                input = Console.ReadLine().ToLower();
            }

            input = Console.ReadLine().ToLower();

            double cost = 0;
            double totalCost = 0;
            while (input != "end")
            {
                switch (input)
                {
                    case "nuts":
                        cost = 2;
                        break;
                    case "water":
                        cost = 0.7;
                        break;
                    case "crisps":
                        cost = 1.5;
                        break;
                    case "soda":
                        cost = 0.8;
                        break;
                    case "coke":
                        cost = 1;
                        break;
                    default:
                        Console.WriteLine("Invalid product");
                        input = Console.ReadLine().ToLower();
                        continue;
                }

                if (cost + totalCost <= cash)
                {
                    Console.WriteLine($"Purchased {input}");
                    totalCost += cost;
                }
                else if (cost + totalCost > cash)
                {
                    Console.WriteLine("Sorry, not enough money");
                }

                input = Console.ReadLine().ToLower();
            }

            Console.WriteLine($"Change: {cash - totalCost:f2}");
        }
    }
}

File: Java Fundamentals/BasicSyntaxEx/VendingMachine.java
package BasicSyntaxEx;

import java.util.Scanner;

public class VendingMachine {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double sum = 0d;
        String input = scanner.nextLine();
        while (!input.equals("Start")) {
            double coins = Double.parseDouble(input);
            if (coins == 0.1 || coins == 0.2 || coins == 0.5 || coins == 1 || coins == 2) {
                sum += coins;
            } else {
                System.out.printf("Cannot accept %.2f%n", coins);
            }
            input = scanner.nextLine();
        }

        input = scanner.nextLine();
        while (!input.equals("End")) {

            double price = 0d;

            switch (input) {
                case "Nuts":
                    price = 2.0;
                    break;
                case "Water":
                    price = 0.7;
                    break;
                case "Crisps":
                    price = 1.5;
                    break;
                case "Soda":
                    price = 0.8;
                    break;
                case "Coke":
                    price = 1.0;
                    break;
                default:
                    System.out.println("Invalid product");
                    input = scanner.nextLine();
                    continue;
            }

            if (sum - price >= 0) {
                System.out.printf("Purchased %s%n", input);
                sum -= price;
            } else {
                System.out.println("Sorry, not enough money");
            }

            input = scanner.nextLine();
        }

        System.out.printf("Change: %.2f", sum);
    }
}
